libs/sdk-ui-tests-e2e/cypress/tools/permissions.ts

```diff
@@ -0,0 +1,133 @@
+        const data = groups.map((groupId) => {
+            return {
+                id: groupId,
+                type: "userGroup",
+            };
+        });
+        const body = {
+            data: {
+                id,
+                type: "user",
+                attributes: {
+                    authenticationId: `authId_${id}`,
+                },
+                relationships: {
+                    userGroups: {
+                        data,
+                    },
+                },
+            },
+        };
+        const url = `/api/v1/entities/users`;
+        Api.request("POST", url, body);
+        return this;
+    }
+
+    static createGroup(id: string) {
+        const body = {
+            data: {
+                attributes: {},
+                id,
+                type: "userGroup",
+            },
+        };
+
+        const url = `/api/v1/entities/userGroups`;
+        Api.request("POST", url, body);
+        return this;
+    }
+
+    static deleteGroup(id: string) {
+        const url = `/api/v1/entities/userGroups/${id}`;
+        Api.request("DELETE", url, undefined, true);
+        return this;
+    }
+
+    static switchToUser(id: string) {
+        const body = {
+            data: {
+                id,
+                type: "apiToken",
+                attributes: {},
+            },
+        };
+        Api.request("POST", `/api/v1/entities/users/${id}/apiTokens`, body).then((res: any) => {
+            const body = res.body;
+            expect(res.status).eq(201);
+            const token = body.data.attributes.bearerToken;
+            Api.injectAuthHeader(token);
+        });
+    }
+
+    static switchToDefaultUser() {
+        Api.injectAuthHeader(getTigerAuthToken());
+        return this;
+    }
+
+    static deleteUser(id: string) {
+        const url = `/api/v1/entities/users/${id}`;
+        Api.request("DELETE", url, undefined, true);
+        return this;
+    }
+}
+
+export class DashboardAccess {
+    static assignPermission(workspaceId: string, dashboardId: string, userId: string, permissions: string[]) {
+        const url = `/api/v1/actions/workspaces/${workspaceId}/dashboards/${dashboardId}/managePermissions`;
+        const body = [
+            {
+                assigneeIdentifier: {
+                    id: userId,
+                    type: "user",
+                },
+                permissions,
+            },
+        ];
+
+        // TODO: why issue with content-type
+        // Api.request("POST", url, body);
+
+        cy.request({
+            method: "POST",
+            url: url.startsWith("/") ? `${getHost()}${url}` : url,
+            headers: {
+                Authorization: `BEARER ${getTigerAuthToken()}`,
+                "Content-Type": "application/json",
+            },
+            body,
+        });
+    }
+
+    static assignUserPermissionToWorkspace(workspaceId: string, userId: string, permission: string) {
```

Comment:
We could possibly separate this as it belongs more to something like `WorkspaceAccess` than `DashboardAccess`. Then you could simply rename this method to `assignPermission`.